proto/xudrpc.proto

```diff
@@ -366,6 +373,57 @@ message ListPeersResponse {
   repeated Peer peers = 1 [json_name = "peers"];
 }
 
+message ListSwapDealsRequest {
+  bool all = 1 [json_name = "all"];
```

Comment:
Rather than an `all` request parameter, I think it'd be better to have a numeric `limit` as you're doing in the ListTrades PR.  If it's 0, we return all swaps, otherwise we return the most recent ones up to the limit. We can also have a `failed` flag to return only failed swaps, otherwise we return only successful swaps. Let me know what you think.

Reply:
@sangaman I agree but why would we want to return only successful or failed swap?
Shouldn't `failed` just indicate if we should include failed swaps in the response?